### ISSUE TITLE
Phalcon\Mvc\Collection support for _id as array

### ISSUE THREAD

**Issue by challet** [phalcon/cphalcon#11134]
When a mongo collection has its _ids made of arrays, for instance the result of a [group](https://docs.mongodb.org/v3.0/reference/operator/aggregation/group/) aggregation, it causes the `Collection::find` method to throw a "Array to string conversion" notice on the [iterator_to_array function call](https://github.com/phalcon/cphalcon/blob/master/phalcon/mvc/collection.zep#L432) ~~then a "You cannot serialize or unserialize PDO instances" PDOException~~ (**edit** : not part of the case).

How would it work with the [use_keys parameter](http://php.net/iterator_to_array#use_keys) set to false ? since the key doesn't seem to be used later in the code. By the way the `Collection::findFirst` call, which returns just before this operation, handles it correctly and makes it available as an array under `$result->_id`.

**Comment by challet** [comment]
script to test the described beahvior

```php
<?php 
use Phalcon\Mvc\Collection as Collection;

class TestDocument extends Collection {
  
  public function getSource() {
    return "test";
  }
  
  public function initialize() {
    $this->useImplicitObjectIds(false);
  }

}

// ------------------------------------------
$doc = new TestDocument();
$doc->_id = array(
  'a' => 1,
  'b' => 1
);
$doc->save();

var_dump(TestDocument::find());
```

it leads to : 
```
Notice: Array to string conversion in test.php on line 8
Call Stack:
[...]
    0.0459     378464   6. Phalcon\Mvc\Collection::find() test.php:8
    0.0461     379880   7. Phalcon\Mvc\Collection::_getResultset(array (), class \TestDocument { public $_id = NULL; protected $_dependencyInjector = class Phalcon\Di\FactoryDefault\Cli { protected $_services = array ('router' => class Phalcon\Di\Service { ... }, 'dispatcher' => class Phalcon\Di\Service { ... }, 'modelsManager' => class Phalcon\Di\Service { ... }, 'modelsMetadata' => class Phalcon\Di\Service { ... }, 'filter' => class Phalcon\Di\Service { ... }, 'escaper' => class Phalcon\Di\Service { ... }, 'annotations' => class Phalcon\Di\Service { ... }, 'security' => class Phalcon\Di\Service { ... }, 'eventsManager' => class Phalcon\Di\Service { ... }, 'transactionManager' => class Phalcon\Di\Service { ... }, 'config' => class Phalcon\Di\Service { ... }, 'db' => class Phalcon\Di\Service { ... }, 'mongo' => class Phalcon\Di\Service { ... }, 'collectionManager' => class Phalcon\Di\Service { ... }, 'queue' => class Phalcon\Di\Service { ... }, 'translater' => class Phalcon\Di\Service { ... }); protected $_sharedInstances = array ('config' => class Phalcon\Config\Adapter\Json { ... }, 'router' => class Phalcon\Cli\Router { ... }, 'dispatcher' => class Phalcon\Cli\Dispatcher { ... }, 'Fai\Tasks\CompetitionTask' => class Fai\Tasks\CompetitionTask { ... }, 'collectionManager' => class Phalcon\Mvc\Collection\Manager { ... }, 'mongo' => class MongoDB { ... }); protected $_freshInstance = FALSE; protected $_eventsManager = NULL }; protected $_modelsManager = class Phalcon\Mvc\Collection\Manager { protected $_dependencyInjector = class Phalcon\Di\FactoryDefault\Cli { protected $_services = array (...); protected $_sharedInstances = array (...); protected $_freshInstance = FALSE; protected $_eventsManager = NULL }; protected $_initialized = array ('\TestDocument' => class \TestDocument { ... }); protected $_lastInitialized = class \TestDocument { public $_id = array (...); protected $_dependencyInjector = class Phalcon\Di\FactoryDefault\Cli { ... }; protected $_modelsManager = ...; protected $_source = NULL; protected $_operationMade = 2; protected $_connection = class MongoDB { ... }; protected $_errorMessages = array (...); protected $_skipped = FALSE }; protected $_eventsManager = class Phalcon\Events\Manager { protected $_events = NULL; protected $_collect = FALSE; protected $_enablePriorities = FALSE; protected $_responses = NULL }; protected $_customEventsManager = NULL; protected $_connectionServices = NULL; protected $_implicitObjectsIds = array ('\TestDocument' => FALSE); protected $_behaviors = NULL }; protected $_source = NULL; protected $_operationMade = 0; protected $_connection = class MongoDB { public $w = 1; public $wtimeout = 10000 }; protected $_errorMessages = NULL; protected $_skipped = FALSE }, class MongoDB { public $w = 1; public $wtimeout = 10000 }, FALSE) test.php:8
    0.0462     381232   8. iterator_to_array(class MongoCursor {  }) test.php:8
    0.0465     382800   9. MongoCursor->key() test.php:8


array(1) {
  [0] =>
  class TestDocument#50 (8) {
    public $_id =>
    array(2) {
      'a' =>
      int(1)
      'b' =>
      int(1)
    }
    ...
```


Despite the thrown notice, the result seems to be the one expected : the `_id`field is retrieved with it's right value.

**Comment by denisbulakh** [comment]
Confirm the issue. But with 2.0.9 version and php 5.5.9 results are wrong. Only one result returned even if there are much more.

**Comment by challet** [comment]
fixed with #11840
